OloEngine/src/OloEngine/Renderer/WaterSurface.cpp

```
@@ -0,0 +1,173 @@
+
+    glm::vec3 SampleDisplacement(const Params& params, glm::vec2 baseXZ, f32 rawTime)
+    {
+        // The water vertex shader evaluates `time = u_WaveParams.x * u_WaveParams.y`
+        // (= Time * WaveSpeed) before calling sumGerstnerWaves — match it exactly.
+        const f32 time = rawTime * params.m_WaveSpeed;
+        const glm::vec3 basePos(baseXZ.x, params.m_PlaneHeight, baseXZ.y);
+        return SumGerstnerDisplacement(basePos, time, params.m_WaveDir0, params.m_WaveDir1,
+                                       params.m_WaveFrequency, params.m_WaveAmplitude);
+    }
+
+    f32 SampleHeight(const Params& params, glm::vec2 queryXZ, f32 rawTime)
```

Cppcheck (2.20.0)

[style] The function 'SampleHeight' is never used.

(unusedFunction)